fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/OneOfThemWithComparison.java

```diff
@@ -128,7 +128,8 @@ public int hashCode() {
     }
 
     @Override
-    public int planHash() {
-        return getComparison().planHash();
+    public int planHash(PlanHashKind hashKind) {
```

Comment:
This has a problem. I don't understand why `fieldName` is not part of equals or the regular hash code. If it should be then, `planHash(...)` should incorporate that as well.